code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,517 @@
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with greater fortitude."
+	duration = 10 MINUTES
+	cost = 2
+	number_of_required_pylons = 2
+	radial_icon = "health_m"
+
+/datum/hivebuff/extra_life/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth *= 1.1
+
+/datum/hivebuff/extra_life/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.maxHealth = initial(xeno.caste.max_health)
+
+/datum/hivebuff/game_ender_caste
+	name = "Boon of Destruction"
+	desc = "A huge behemoth of a Xenomorph which can tear its way through defences and flesh alike."
+	tier = HIVEBUFF_TIER_MAJOR
+	is_unique = TRUE
+	is_reusable = TRUE
+	cost = 0
+	special_fail_message = "Only one hatchery may exist at a time."
+	number_of_required_pylons = 2
+	roundtime_to_enable = 1 HOURS + 50 MINUTES
+
+/datum/hivebuff/game_ender_caste/handle_special_checks()
+	if(hive.has_hatchery)
+		return FALSE
+
+	return TRUE
+
+/datum/hivebuff/game_ender_caste/on_engage(obj/effect/alien/resin/special/pylon/purchased_pylon)
+	var/turf/spawn_turf
+	for(var/turf/potential_turf in orange(5, purchased_pylon))
+		var/failed = FALSE
+		for(var/x_offset in 0 to 3)
+			for(var/y_offset in 0 to 3)
+				var/turf/turf_to_check = locate(potential_turf.x + x_offset, potential_turf.y + y_offset, potential_turf.z)
+
+				if(turf_to_check.density)
+					failed = TRUE
+					break
+				var/area/target_area = get_area(turf_to_check)
+				if(target_area.flags_area & AREA_NOTUNNEL)
+					failed = TRUE		
+					break			
+		if(!failed)
+			spawn_turf = potential_turf
+			break
+
+	if(!spawn_turf)
+		engage_failure_message = "Unable to find a viable spawn point for the Destroyer"
+		return FALSE
+
+	new /obj/effect/alien/resin/destroyer_cocoon(spawn_turf)
+
+	return TRUE
+
+/datum/hivebuff/defence
+	name = "Boon of Defence"
+	desc = "Increases all xenomorph armour by 5% for 5 minutes"
+	tier = HIVEBUFF_TIER_MINOR
+
+	engage_flavourmessage = "The Queen has imbued us with greater chitin."
+	duration = 5 MINUTES
+	number_of_required_pylons = 1
+	radial_icon = "shield"
+
+/datum/hivebuff/defence/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection += 0.05 * xeno.armor_deflection
+
+/datum/hivebuff/defence/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection = initial(xeno.armor_deflection)
+
+/datum/hivebuff/defence/major
+	name = "Major Boon of Defence"
+	desc = "Increases all xenomorph armour by 10% for 10 minutes"
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with even greater chitin."
+	duration = 10 MINUTES
+	cost = 2
+	number_of_required_pylons = 2
+	radial_icon = "shield_m"
+
+/datum/hivebuff/defence/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection += 0.1 * xeno.armor_deflection
+
+/datum/hivebuff/defence/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection = initial(xeno.armor_deflection)
```

Comment:
These modifications should be applied to `armor_modifier` as + / - because you will lose the armor_deflection
 once `recalculate_armor` is called.

Also consider `armor_explosive_buff`

Reply:
Also of note: multiplying against 0 is still 0. Needs to be additive.